oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2TokenAttributes.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * A domain object that wraps around the attributes in an {@link org.springframework.security.core.Authentication}
+ * token.
+ *
+ * @author Clement Ng
+ * @since 5.2
+ */
+public final class OAuth2TokenAttributes {
+	private Map<String, Object> attributes;
+
+	public OAuth2TokenAttributes(Map<String, Object> attributes) {
+		this.attributes = Collections.unmodifiableMap(attributes);
+	}
+
+	public Map<String, Object> getAttributes() {
+		return attributes;
+	}
+
+	public <A> A getAttribute(String name) {
```

Comment:
Would you please add some simple Javadoc to each public method you are introducing?